### PR TITLE
Minor: Fix possible NPE.

### PR DESCRIPTION
**Pull request by nk1506**
# Description

Fix for possible NPE. 

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

Please delete options that are not relevant.

- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings

### CODE REVIEW

File: polaris-core/src/main/java/io/polaris/core/persistence/resolver/PolarisResolutionManifest.java [hunk 1]
```diff
@@ -403,7 +403,8 @@ public PolarisResolvedPathWrapper getResolvedTopLevelEntity(
     if (resolvedCacheEntry == null) {
       return null;
     }
-    return new PolarisResolvedPathWrapper(
-        List.of(getResolvedRootContainerEntity(), new ResolvedPolarisEntity(resolvedCacheEntry)));
+
+    ResolvedPolarisEntity resolvedRootContainerEntity = getResolvedRootContainerEntity();
+    return resolvedRootContainerEntity == null ? new PolarisResolvedPathWrapper(List.of(new ResolvedPolarisEntity(resolvedCacheEntry))):new PolarisResolvedPathWrapper(List.of(resolvedRootContainerEntity, new ResolvedPolarisEntity(resolvedCacheEntry))) ;
```
Comment (ebyhr) [inline polaris-core/src/main/java/io/polaris/core/persistence/resolver/PolarisResolutionManifest.java:408]: This line looks too long. How about wrapping like below? 
```java
    return resolvedRootContainerEntity == null
            ? new PolarisResolvedPathWrapper(List.of(new ResolvedPolarisEntity(resolvedCacheEntry)))
            : new PolarisResolvedPathWrapper(List.of(resolvedRootContainerEntity, new ResolvedPolarisEntity(resolvedCacheEntry))) ;
```

Comment (nk1506) [reply polaris-core/src/main/java/io/polaris/core/persistence/resolver/PolarisResolutionManifest.java]: I was hoping build will fail with spotless check.

File: polaris-core/src/main/java/io/polaris/core/persistence/ResolvedPolarisEntity.java [hunk 2]
```diff
@@ -45,6 +46,7 @@ public ResolvedPolarisEntity(
   }
 
   public ResolvedPolarisEntity(EntityCacheEntry cacheEntry) {
```
Comment (snazy) [inline polaris-core/src/main/java/io/polaris/core/persistence/ResolvedPolarisEntity.java:48]:
```suggestion
  public ResolvedPolarisEntity(@NotNull EntityCacheEntry cacheEntry) {
```

File: polaris-core/src/main/java/io/polaris/core/persistence/ResolvedPolarisEntity.java [hunk 3]
```diff
@@ -45,6 +46,7 @@ public ResolvedPolarisEntity(
...
+    Preconditions.checkNotNull(cacheEntry, "An entity cache can't be null.");
```
Comment (snazy) [inline polaris-core/src/main/java/io/polaris/core/persistence/ResolvedPolarisEntity.java:49]: The check is redundant, because recent JVMs throw meaningful NPEs telling that `cacheEntry` is `null` - the proposed message doesn't give any advice to users.

File: polaris-service/src/main/java/io/polaris/service/catalog/BasePolarisCatalog.java [hunk 4]
```diff
@@ -1320,7 +1320,7 @@ private void validateMetadataFileInTableDir(
     return storageInfoEntity;
   }
 
-  private class BasePolarisViewOperations extends BaseViewOperations {
+  protected class BasePolarisViewOperations extends BaseViewOperations {
```
Comment (snazy) [inline polaris-service/src/main/java/io/polaris/service/catalog/BasePolarisCatalog.java:1323]: Why this change?

File: polaris-core/src/main/java/io/polaris/core/persistence/resolver/PolarisResolutionManifest.java [hunk 5]
```diff
@@ -403,7 +403,11 @@ public PolarisResolvedPathWrapper getResolvedTopLevelEntity(
     if (resolvedCacheEntry == null) {
       return null;
     }
-    return new PolarisResolvedPathWrapper(
-        List.of(getResolvedRootContainerEntity(), new ResolvedPolarisEntity(resolvedCacheEntry)));
+
+    ResolvedPolarisEntity resolvedRootContainerEntity = getResolvedRootContainerEntity();
```
Comment (snazy) [inline polaris-core/src/main/java/io/polaris/core/persistence/resolver/PolarisResolutionManifest.java:407]: Looks like there are more calls to `getResolvedRootContainerEntity()` that can fail with a direct or indirect NPE.
@collado-mike WDYT?

File: polaris-service/src/main/java/io/polaris/service/catalog/BasePolarisCatalog.java (hunk 4 above)
Comment (nk1506) [reply polaris-service/src/main/java/io/polaris/service/catalog/BasePolarisCatalog.java]: It was violating the visibility of [BasePolarisViewOperations](https://github.com/polaris-catalog/polaris/blob/5c78eefe0f029b5140d41f74c3c4fc1710fc5e25/polaris-service/src/main/java/io/polaris/service/catalog/BasePolarisCatalog.java#L707) .

File: polaris-service/src/main/java/io/polaris/service/catalog/BasePolarisCatalog.java [hunk 6]
```diff
@@ -704,7 +705,7 @@ public List<TableIdentifier> listViews(Namespace namespace) {
   }
 
   @Override
-  protected BasePolarisViewOperations newViewOps(TableIdentifier identifier) {
```
Comment (eric-maynard) [inline polaris-service/src/main/java/io/polaris/service/catalog/BasePolarisCatalog.java:707]: This seems unrelated to the NPE fix

Comment (nk1506) [reply polaris-service/src/main/java/io/polaris/service/catalog/BasePolarisCatalog.java:707]: Yeah, Let me remove this and fix with separate PR.

**Review by RussellSpitzer — APPROVED** [review]
Looks like all feedback was applied here, Approving and letting tests run. I'll let the original reviewers merge.

**Comment by takidau** [comment]
This has been sitting for a while, I'm just going to merge.
